build/vsts_build.yaml

```diff
@@ -807,12 +839,22 @@ jobs:
       downloadPath: "$(Build.Repository.LocalPath)/artifacts"
 
   - task: ShellScript@2
-    displayName: "Run Tests"
+    displayName: "Run Tests (continue on error)"
```

Comment:
Are the lines 810-857 the same as 759-774 ?

Reply:
Basically yes. Lines 759-774 are running tests on Linux, lines 810-857 are running tests on Mac.

Reply:
can you explain why we're doing this? 

No variables in continue on error?

Reply:
correct: https://developercommunity.visualstudio.com/content/problem/1187733/variablesexpressions-dont-work-with-continueonerro.html

I think variablies are just syntactic sugar for a less verbose runtime expression, but `continueOnError`, and some other yaml values, do not evaluate runtime expressions. Compile time expressions are evaluated **before** build variables are loaded, hence neither compile time expressions or parameters can be used to change behaviour per build definition in AzDO. I'll probably delete it soon, but I did create a [test repo](https://github.com/zivkan/azure-pipelines-continueOnError-test) where I could quickly iterate on every "creative" idea I could think of, but AzDO's yaml parsing/evaluation is too limited.

The only other alternative I could think of was to make our build yml a template, and have different yml entry points, one for official and one for private. That way we don't have to duplicate anything and can use compile time expressions or parameters. But if we did that, if we ever wanted to service an old release branch, we'd have to spend time fixing the build files.